tests/ui/traits/non_lifetime_binders/non-lifetime-binders-issue-119067.rs

```diff
@@ -0,0 +1,18 @@
+// check-fail
+
+#![allow(incomplete_features)]
+#![feature(non_lifetime_binders)]
+#![feature(closure_lifetime_binder)]
+
+trait Trait {}
+
+trait Trait2
+    where for <T: Trait> ():{}
+//~^ ERROR bounds cannot be used in this context
+
+fn main() {
+    let _ = for<T: Trait> || {};
```

Comment:
```suggestion
    let _ = for<T: Trait> -> () || {};
```

This way, you can suppress *implicit types are not allowed …* error.

Reply:
Done. Thank you.